simple/app.go

```diff
@@ -0,0 +1,334 @@
+package simple
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+
+	"github.com/grafana/grafana-app-sdk/app"
+	"github.com/grafana/grafana-app-sdk/k8s"
+	"github.com/grafana/grafana-app-sdk/operator"
+	"github.com/grafana/grafana-app-sdk/resource"
+)
+
+var _ app.Provider = &AppProvider{}
+
+type Converter k8s.Converter
+
+// AppProvider is a simple implementation of app.Provider which returns AppManifest when Manifest is called,
+// and calls NewAppFunc when NewApp is called.
+type AppProvider struct {
+	AppManifest app.Manifest
+	NewAppFunc  func(config app.Config) (app.App, error)
+}
+
+// Manifest returns the AppManifest in the AppProvider
+func (a *AppProvider) Manifest() app.Manifest {
+	return a.AppManifest
+}
+
+// NewApp calls NewAppFunc and returns the result
+func (a *AppProvider) NewApp(settings app.Config) (app.App, error) {
+	return a.NewAppFunc(settings)
+}
+
+// NewAppProvider is a convenience method for creating a new AppProvider
+func NewAppProvider(manifest app.Manifest, newAppFunc func(cfg app.Config) (app.App, error)) *AppProvider {
+	return &AppProvider{
+		AppManifest: manifest,
+		NewAppFunc:  newAppFunc,
+	}
+}
+
+var (
+	_ app.App = &App{}
+)
+
+// App is a simple, opinionated implementation of resource.App.
+// It must be created with NewApp to be valid.
+type App struct {
+	informerController *operator.InformerController
+	operator           *operator.Operator
+	clientGenerator    resource.ClientGenerator
+	kinds              map[string]AppManagedKind
+	internalKinds      map[string]resource.Kind
+	cfg                AppConfig
+	converters         map[string]Converter
+}
+
+// AppConfig is the configuration used by App
+type AppConfig struct {
+	Name           string
+	KubeConfig     rest.Config
+	InformerConfig AppInformerConfig
+	ManagedKinds   []AppManagedKind
+	Converters     map[schema.GroupKind]Converter
```

Comment:
Do you think this could be part of the `AppManagedKind` instead?

Reply:
I tried messing around with it, the problem being that `AppManagedKind` is a GroupVersionKind, and a converter is for a GroupKind. We'd have to restructure the way that conversion is handled, and I didn't have a good solution for it that didn't feel overly complicated. I'd still like to revisit this in the future.